scaffolding-chef-inspec/lib/scaffolding.ps1

```diff
@@ -24,7 +24,12 @@ function Invoke-DefaultBefore {
     }
 
     if((Select-String -Path "$PLAN_CONTEXT/../inspec.yml" -Pattern "compliance: ")){
-        Invoke-ComplianceLogin
+        if(!$(Test-Path "~/.inspec/compliance/config.json")){
```

Comment:
This is a bit tricky -- I'm spinning up some tests on this, but I'm not sure this will work in the Windows docker studio. This might still work outside the studio context. In other words, we have no guarantee that this file exists, and it's a bit of friction for the user to "remember" to do this chore beforehand. We want the build to automatically do all of this stuff.

I wonder if this can lead to some surprising behavior? Does this create a split in the workflow? Does this mean that the user needs to login beforehand? Or should we ALWAYS make sure that credentials are in the plan?

Perhaps the right way to go here is to check if `$scaffold_automate_*` variables are set, and if they are, then login, else throw an appropriate error?

Reply:
The tricky bit here is that we need a way to "bootstrap" the compliance login so to speak. So we can't start with any assumption that the user has already logged in for the first time. The super tricky part is the token auth required. Is it a good idea to store this token in the plan?

We are doing this with the automate login for now, and it seems OK -- as long as the permissions for that token are set to data_collector only. Does IAMv2 allow us to keep the permissions of this token small enough that it isn't a concern? (IE: read only access to download profiles from Chef Automate).

I think this might be OK for now, but I can also see it being easy to hamstring yourself by accidentally using a token with too broad permissions.

Reply:
Building in the docker studio for consuming packages is busted right now, I discovered. See #85

Reply:
So my thought had been that if you have a `compliance: ` line in you `inspec.yml` it will fail to build the profile so this just exits but I realize I have removed the warning message so I will add that back in.